src/Http/Controllers/Filters/Search.php

```diff
@@ -23,6 +23,13 @@ public static function apply(Controller $controller, Builder|Relation $query): v
         $model = app($controller->model);
 
         $searchValue = request()->input('search_value');
+        
+        if (empty($searchValue)) {
+            return;
+        }
```

Comment:
This check already exists at line 35. Maybe move the `mutateSearchValue` call below that line.

Reply:
I intentinally did this because although we are checking the same variable twice, we are not necessarily checking the same value twice.

If `$searchValue` is `null` on the first check (line 27), the `$controller->mutateSearchValue($searchValue)` will crash as it expects a string.

If we do not recheck it on line 35, it might have been mutated into an empty string.

The only alternative could be a try/catch arount the `$controller->mutateSearchValue($searchValue)`. Would you prefer that?

Reply:
`mutateSearchValue` must return a string. If it's an empty string, that's either intentional or by accident, either way, it's on the developer who implemented that function.
We only need to check if the input search value is empty. So imo we can just move the `mutateSearchValue` call below line 37.

Reply:
Fair point, moved it below the if-statement.